File: frontend/src/types/projectTag.ts
// GET レスポンス用（FastAPI の ProjectTagResponse に対応）
export type ProjectTag = {
  id: number;
  project_id: number;
  name: string;
  created_at: string;
};

// POST リクエスト用（FastAPI の ProjectTagCreate に対応）
export type ProjectTagCreate = {
  project_id: number;
  name: string;
};

File: backend/app/schemas/project_tag.py
from pydantic import BaseModel
from datetime import datetime


class ProjectTagCreate(BaseModel):
    """POST リクエスト時に受け取るデータ"""
    project_id: int
    name: str


class ProjectTagResponse(BaseModel):
    """レスポンスとして返すデータ"""
    id: int
    project_id: int
    name: str
    created_at: datetime

    class Config:
        from_attributes = True
